.github/workflows/repro_check.yml

```diff
@@ -0,0 +1,79 @@
+name: Build and Diff Projects
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+      - reproducible
+
+jobs:
+  build_and_compare:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.x'  # Adjust to the version you need
+
```

Comment:
```suggestion
```
can't we use the default python preinstalled in github actions runners?
I think this is what we do when we run `x.py` usually.

`actions/setup-python` is not used in the codebase